packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -1094,11 +1096,38 @@ class EditableText extends StatefulWidget {
   /// The cursor will draw under the text. The cursor width will extend
   /// to the right of the boundary between characters for left-to-right text
   /// and to the left for right-to-left text. This corresponds to extending
-  /// downstream relative to the selected position. Negative values may be used
-  /// to reverse this behavior.
+  /// downstream relative to the selected position.
+  /// 
+  /// See also:
+  /// 
+  ///  [caretMargin], [cursorPadding] for more control of your cursor layout.
+  /// 
+  ///    Flutter.
   /// {@endtemplate}
   final double cursorWidth;
 
+  /// {@template flutter.widgets.editableText.cursorPadding}
+  /// How many pixels between the end of the cursor and the end of the text including caretMargin.
+  /// 
+  /// Defaults to 1.0.
+  /// 
+  /// See also:
+  /// 
+  ///  [cursorWidth], [caretMargin] for more control of your cursor layout.
+  /// 
+  /// {@endtemplate}
+  final double cursorPadding;
+
+  /// {@template flutter.widgets.editableText.caretMargin}
+  /// How many pixels will be added after the last character.
```

Comment:
To me the documentation is a bit confusing without the right context:
1. Why are we adding pixels after the last character? What does this control exactly?
2. Where are we adding the pixels to if specified?
3. What happens if `caretMargin` is null? 
4. What's the difference between this and `cursorPadding`?

Reply:
Documentation edited ,
The pixels after the last character is designed/needed to contain the cursor,  and it is added after the last character in the text.
if the `caretMargin`  is null this will fall back to the old default behavior which calculates the `caretMargin`  to be  `cursorWidth`  + the a gap of 1 pixel which now is the `cursorPadding` which defaults to 1.0. to work with the defaluts and does not affect the existing tests.
hope that these answer communicated weill in the documentations now . please review the changes.